### ISSUE TITLE
Curses applications are not working with the Mingw32 Python

### ISSUE THREAD

**Issue by dobairoland** [msys2/MINGW-packages#5823]
Code to repoduce:
```python
import curses

stdscr = curses.initscr()
curses.noecho()
curses.cbreak()
stdscr.keypad(1)


curses.nocbreak()
stdscr.keypad(0)
curses.echo()
curses.endwin()
```
Environment:
```sh
export TERM=xterm-256color
export TERMINFO=/mingw32/share/terminfo
```

`/mingw32/bin/python3` and `/mingw32/bin/python2` are bind together with `/mingw32/share/terminfo` through package dependencies between  `mingw-w64-i686-python2` (&3) and `mingw-w64-i686-ncurses`.

The error output:
```python
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    curses.cbreak()
_curses.error: cbreak() returned ERR
```

**Comment by lazka** [comment]
Not much experience with curses, but `winpty python3 foo.py` seems to make it work.

**Comment by dobairoland** [comment]
Hi @lazka. Thank you for looking at this.

Yes, after I update to Python version 3.8.0 then `winpty` seems to help with that simple example. However, I still have issues with a more complex example. I'm trying to run this code
https://github.com/ulfalizer/Kconfiglib/blob/master/menuconfig.py which works perfectly fine on Linux, MacOs and Windows command line. I'm experiencing these issue only in Mingw32.

The failure with `winpty` is different: 
`_curses.error: use_default_colors() returned ERR`  for both `xterm` and `xterm-256color`.

**Comment by lazka** [comment]
Do you mean it works with the official Python on Windows?

Maybe we should just disable curses then.

**Comment by dobairoland** [comment]
> Do you mean it works with the official Python on Windows?

Yes it does work with this additional package: https://pypi.org/project/windows-curses/

The MSYS Pythons are also working natively (without any supporting external package). The problem is only with the MIGW pythons. Our customers are relying on that and that is why I reported this issue.

**Comment by ulfalizer** [comment]
Hmm... wonder if I should ignore errors from `use_default_colors()` and print a warning instead. Apparently checking `has_colors()` isn't enough.

Only side effect would be that you can't use default (user-set) colors, for glitzy transparency stuff and the like.

**Comment by ulfalizer** [comment]
@dobairoland 
Is removing the `use_default_colors()` enough to get it running? Don't have MINGW handy atm.

**Comment by ulfalizer** [comment]
Doesn't seem like it. Get a non-Python crash with exit status 58 when it tries to read a character with `getch()` instead, whatever that means.

Hardcoding a return from `getch()` gives another crash somewhere else with nothing drawn.

Doesn't seem super stable...

I'll add a fix to ignore errors from `use_default_colors()` at least. Think it's the right thing to do for exotic terminals regardless.

**Comment by dobairoland** [comment]
@ulfalizer  Thanks for looking at this as well.

> Is removing the use_default_colors() enough to get it running?

No. I'm getting another error after that:
`_curses.error: init_pair() returned ERR`

**Comment by ulfalizer** [comment]
> No. I'm getting another error after that:
> `_curses.error: init_pair() returned ERR`

Wonder if that's the old brightwhite-in-aquatic-theme thing. Fixed it upstream.

Not sure what's going on otherwise.
